network/endpoint_windows.go

```diff
@@ -44,6 +46,21 @@ func ConstructEndpointID(containerID string, netNsPath string, ifName string) (s
 
 // newEndpointImpl creates a new endpoint in the network.
 func (nw *network) newEndpointImpl(epInfo *EndpointInfo) (*endpoint, error) {
+	// Check if the netNsPath is a valid GUID to decide on HNSv1 or HNSv2
```

Comment:
I think we can move this whole block that checks whether to do newEndpointImplHnsV2 or newEndpointImplHnsV1 into a single func